diff --git a/global/src/ga.h b/global/src/ga.h
index a5b11dc9..d65f85a3 100644
--- a/global/src/ga.h
+++ b/global/src/ga.h
@@ -22,7 +22,6 @@ extern void          GA_Add_constant_patch(int g,int *lo,int *hi,void *alpha);
 extern void          GA_Add_diagonal(int g_a, int g_v);
 extern void          GA_Add(void *alpha, int g_a, void* beta, int g_b, int g_c); 
 extern int           GA_Allocate(int g_a);
-extern int           GA_Deallocate(int g_a);
 extern int           GA_Assemble_duplicate(int g_a, char *name, void *ptr);
 extern void          GA_Brdcst(void *buf, int lenbuf, int root);
 extern SingleComplex GA_Cdot(int g_a, int g_b); 
@@ -38,6 +37,7 @@ extern int           GA_Compare_distr(int g_a, int g_b);
 extern void          GA_Copy(int g_a, int g_b); 
 extern int           GA_Create_handle(void);
 extern int           GA_Create_mutexes(int number);
+extern int           GA_Deallocate(int g_a); 
 extern double        GA_Ddot(int g_a, int g_b); 
 extern void          GA_Destroy(int g_a);
 extern int           GA_Destroy_mutexes(void);
@@ -110,10 +110,9 @@ extern int           GA_Pgroup_absolute_id(int pgroup, int pid);
 extern void          GA_Pgroup_brdcst(int grp, void *buf, int lenbuf, int root);
 extern void          GA_Pgroup_cgop(int grp, SingleComplex x[], int n, char *op);
 extern int           GA_Pgroup_create(int *list, int count);
-extern int           GA_Pgroup_duplicate(int grp);
-extern int           GA_Pgroup_self();
 extern int           GA_Pgroup_destroy(int grp);
 extern void          GA_Pgroup_dgop(int grp, double x[], int n, char *op);
+extern int           GA_Pgroup_duplicate(int grp);
 extern void          GA_Pgroup_fgop(int grp, float x[], int n, char *op);   
 extern int           GA_Pgroup_get_default(void);
 extern int           GA_Pgroup_get_mirror(void);
@@ -123,6 +122,7 @@ extern void          GA_Pgroup_lgop(int grp, long x[], int n, char *op);
 extern void          GA_Pgroup_llgop(int grp, long long x[], int n, char *op);
 extern int           GA_Pgroup_nnodes(int grp_id);
 extern int           GA_Pgroup_nodeid(int grp_id);
+extern int           GA_Pgroup_self();
 extern void          GA_Pgroup_set_default(int p_handle);
 extern int           GA_Pgroup_split(int grp_id, int num_group);
 extern int           GA_Pgroup_split_irreg(int grp_id, int color);
@@ -145,6 +145,8 @@ extern void          GA_Scan_copy(int g_a, int g_b, int g_sbit, int lo, int hi);
 extern void          GA_Set_array_name(int g_a, char *name);
 extern void          GA_Set_block_cyclic(int g_a, int dims[]);
 extern void          GA_Set_block_cyclic_proc_grid(int g_a, int block[], int proc_grid[]);
+extern void          GA_Set_tiled_proc_grid(int g_a, int block[], int proc_grid[]);
+extern void          GA_Set_tiled_irreg_proc_grid(int g_a, int mapc[], int nblocks[], int proc_grid[]);
 extern void          GA_Set_chunk(int g_a, int chunk[]);
 extern void          GA_Set_data(int g_a, int ndim, int dims[], int type);
 extern void          GA_Set_debug(int flag);
@@ -154,14 +156,12 @@ extern void          GA_Set_ghosts(int g_a, int width[]);
 extern void          GA_Set_irreg_distr(int g_a, int map[], int block[]);
 extern void          GA_Set_irreg_flag(int g_a, int flag);
 extern void          GA_Set_memory_limit(size_t limit);
+extern void          GA_Set_memory_dev(int g_a, char *device);
 extern void          GA_Set_pgroup(int g_a, int p_handle);
 extern void          GA_Set_restricted(int g_a, int list[], int size);
 extern void          GA_Set_restricted_range(int g_a, int lo_proc, int hi_proc);
 extern void          GA_Set_property(int g_a, char *property);
-extern void          GA_Set_tiled_proc_grid(int g_a, int block[], int proc_grid[]);
-extern void          GA_Set_tiled_irreg_proc_grid(int g_a, int mapc[], int nblocks[], int proc_grid[]);
 extern void          GA_Unset_property(int g_a);
-extern void          GA_Set_memory_dev(int g_a, char *device);
 extern void          GA_Sgemm(char ta, char tb, int m, int n, int k, float alpha, int g_a, int g_b, float beta, int g_c );
 extern void          GA_Shift_diagonal(int g_a, void *c);
 extern int           GA_Solve(int g_a, int g_b);
@@ -183,7 +183,6 @@ extern int           GA_Uses_ma(void);
 extern int           GA_Uses_proc_grid(int g_a);
 extern int           GA_Valid_handle(int g_a);
 extern int           GA_Verify_handle(int g_a);
-extern void          GA_Version(int *major, int *minor, int *patch);
 extern double        GA_Wtime(void);
 extern DoubleComplex GA_Zdot(int g_a, int g_b); 
 extern void          GA_Zgop(DoubleComplex x[], int n, char *op);
@@ -199,7 +198,6 @@ extern void          NGA_Access(int g_a, int lo[], int hi[], void *ptr, int ld[]
 extern void          NGA_Acc(int g_a, int lo[], int hi[],void* buf,int ld[],void* alpha);
 extern void          NGA_Add_patch(void * alpha, int g_a, int alo[], int ahi[], void * beta,  int g_b, int blo[], int bhi[], int g_c, int clo[], int chi[]);
 extern int           NGA_Allocate(int g_a);
-extern int           NGA_Deallocate(int g_a);
 extern void          NGA_Alloc_gatscat_buf(int nelems);
 extern SingleComplex NGA_Cdot_patch(int g_a, char t_a, int alo[], int ahi[], int g_b, char t_b, int blo[], int bhi[]);
 extern int           NGA_Compare_distr(int g_a, int g_b); 
@@ -214,6 +212,7 @@ extern int           NGA_Create(int type,int ndim,int dims[], char *name, int ch
 extern int           NGA_Create_irreg_config(int type,int ndim,int dims[],char *name, int block[], int map[], int p_handle);
 extern int           NGA_Create_irreg(int type,int ndim,int dims[],char *name, int block[], int map[]);
 extern int           NGA_Create_handle(void);
+extern int           NGA_Deallocate(int g_a); 
 extern int           NGA_Deregister_type(int type);
 extern void          NGA_Destroy(int g_a);
 extern int           NGA_Destroy_mutexes(void);
@@ -224,7 +223,7 @@ extern int           NGA_Duplicate(int g_a, char* array_name);
 extern void          NGA_Error(char *str, int code);
 extern float         NGA_Fdot_patch(int g_a, char t_a, int alo[], int ahi[], int g_b, char t_b, int blo[], int bhi[]);
 extern void          NGA_Fence(void);
-extern void          NGA_Free_gatscat_buf(void);
+extern void          NGA_Free_gatscat_buf();
 extern void          NGA_Fill(int g_a, void *value);
 extern void          NGA_Fill_patch(int g_a, int lo[], int hi[], void *val);
 extern void          NGA_Gather(int g_a, void *v, int* subsArray[], int n);
@@ -276,24 +275,30 @@ extern void          NGA_NbWait(ga_nbhdl_t* nbhandle);
 extern int           NGA_Ndim(int g_a);
 extern int           NGA_Nnodes(void);
 extern int           NGA_Nodeid(void);
-extern int           NGA_Overlay(int g_a, int g_p);
 extern void          NGA_Periodic_acc(int g_a, int lo[], int hi[],void* buf,int ld[],void* alpha);
 extern void          NGA_Periodic_get(int g_a, int lo[], int hi[], void* buf, int ld[]); 
 extern void          NGA_Periodic_put(int g_a, int lo[], int hi[], void* buf, int ld[]); 
+extern int           NGA_Overlay(int g_a, int g_p);
 extern int           NGA_Pgroup_absolute_id(int pgroup, int pid);
 extern int           NGA_Pgroup_create(int *list, int count);
-extern int           NGA_Pgroup_duplicate(int grp);
-extern int           NGA_Pgroup_self();
 extern int           NGA_Pgroup_destroy(int grp);
+extern void          NGA_Pgroup_dgop(int grp, double x[], int n, char *op);
+extern int           NGA_Pgroup_duplicate(int grp);
+extern void          NGA_Pgroup_fgop(int grp, float x[], int n, char *op);   
 extern int           NGA_Pgroup_get_default(void);
 extern int           NGA_Pgroup_get_mirror(void);
 extern int           NGA_Pgroup_get_world(void);
+extern void          NGA_Pgroup_igop(int grp, int x[], int n, char *op);
+extern void          NGA_Pgroup_lgop(int grp, long x[], int n, char *op);
+extern void          NGA_Pgroup_llgop(int grp, long long x[], int n, char *op);
 extern int           NGA_Pgroup_nnodes(int grp_id);
 extern int           NGA_Pgroup_nodeid(int grp_id);
+extern int           NGA_Pgroup_self();
 extern void          NGA_Pgroup_set_default(int p_handle);
 extern int           NGA_Pgroup_split(int grp_id, int num_group);
 extern int           NGA_Pgroup_split_irreg(int grp_id, int color);
 extern void          NGA_Pgroup_sync(int grp_id);
+extern void          NGA_Pgroup_zgop(int grp, DoubleComplex x[], int n, char *op);
 extern void          NGA_Print_patch(int g_a, int lo[], int hi[], int pretty);
 extern void          NGA_Proc_topology(int g_a, int proc, int coord[]);
 extern void          NGA_Put(int g_a, int lo[], int hi[], void* buf, int ld[]); 
@@ -322,6 +327,8 @@ extern void          NGA_Select_elem(int g_a, char* op, void* val, int *index);
 extern void          NGA_Set_array_name(int g_a, char *name);
 extern void          NGA_Set_block_cyclic(int g_a, int dims[]);
 extern void          NGA_Set_block_cyclic_proc_grid(int g_a, int block[], int proc_grid[]);
+extern void          NGA_Set_tiled_proc_grid(int g_a, int block[], int proc_grid[]);
+extern void          NGA_Set_tiled_irreg_proc_grid(int g_a, int mapc[], int nblocks[], int proc_grid[]);
 extern void          NGA_Set_chunk(int g_a, int chunk[]);
 extern void          NGA_Set_data(int g_a, int ndim, int dims[], int type);
 extern void          NGA_Set_debug(int flag);
@@ -329,14 +336,11 @@ extern void          NGA_Set_ghosts(int g_a, int width[]);
 extern void          NGA_Set_irreg_distr(int g_a, int map[], int block[]);
 extern void          NGA_Set_irreg_flag(int g_a, int flag);
 extern void          NGA_Set_memory_limit(size_t limit);
+extern void          NGA_Set_memory_dev(int g_a, char *device);
 extern void          NGA_Set_pgroup(int g_a, int p_handle);
+extern void          NGA_Set_property(int g_a, char *property);
 extern void          NGA_Set_restricted(int g_a, int list[], int size);
 extern void          NGA_Set_restricted_range(int g_a, int lo_proc, int hi_proc);
-extern void          NGA_Set_property(int g_a, char *property);
-extern void          NGA_Set_tiled_proc_grid(int g_a, int block[], int proc_grid[]);
-extern void          NGA_Set_tiled_irreg_proc_grid(int g_a, int mapc[], int nblocks[], int proc_grid[]);
-extern void          NGA_Unset_property(int g_a);
-extern void          NGA_Set_memory_dev(int g_a, char *device);
 extern void          NGA_Strided_acc(int g_a, int lo[], int hi[], int skip[], void* buf, int ld[], void *alpha); 
 extern void          NGA_Strided_get(int g_a, int lo[], int hi[], int skip[], void* buf, int ld[]); 
 extern void          NGA_Strided_put(int g_a, int lo[], int hi[], int skip[], void* buf, int ld[]); 
@@ -344,6 +348,7 @@ extern void          NGA_Sync(void);
 extern void          NGA_Terminate(void);
 extern int           NGA_Total_blocks(int g_a);   
 extern void          NGA_Unlock(int mutex);
+extern void          NGA_Unset_property(int g_a);
 extern void          NGA_Update_ghosts(int g_a);
 extern int           NGA_Update_ghost_dir(int g_a, int dimension, int idir, int flag);
 extern void          NGA_Update_ghosts_nb(int g_a, ga_nbhdl_t *nbhandle);
@@ -351,7 +356,6 @@ extern int           NGA_Uses_ma(void);
 extern int           NGA_Uses_proc_grid(int g_a);
 extern int           NGA_Valid_handle(int g_a);
 extern int           NGA_Verify_handle(int g_a);
-extern void          NGA_Version(int *major, int *minor, int *patch);
 extern double        NGA_Wtime(void);
 extern DoubleComplex NGA_Zdot_patch(int g_a, char t_a, int alo[], int ahi[], int g_b, char t_b, int blo[], int bhi[]);
 extern void          NGA_Zero(int g_a); 
@@ -457,6 +461,10 @@ extern void          NGA_Strided_get64(int g_a, int64_t lo[], int64_t hi[], int6
 extern void          NGA_Strided_put64(int g_a, int64_t lo[], int64_t hi[], int64_t skip[], void* buf, int64_t  ld[]); 
 extern DoubleComplex NGA_Zdot_patch64(int g_a, char t_a, int64_t alo[], int64_t ahi[], int g_b, char t_b, int64_t blo[], int64_t bhi[]);
 extern void          NGA_Zero_patch64(int g_a, int64_t lo[], int64_t hi[]);
+extern int           NGA_Sprs_array_create64(int64_t idim, int64_t jdim, int type);
+extern void          NGA_Sprs_array_add_element64(int s_a, int64_t idx, int64_t jdx, void *val);
+extern void          NGA_Sprs_array_row_distribution64(int s_a, int iproc, int64_t *lo, int64_t *hi);
+extern void          NGA_Sprs_array_column_distribution64(int s_a, int iproc, int64_t *lo, int64_t *hi);
 
 #ifdef __cplusplus
 }
diff --git a/global/src/global.fh.in b/global/src/global.fh.in
index f7820234..dcf88cd9 100644
--- a/global/src/global.fh.in
+++ b/global/src/global.fh.in
@@ -3,7 +3,6 @@
       parameter (ga_max_dim = GA_MAX_DIM)
 !
       logical          ga_allocate
-      logical          ga_deallocate
       complex          ga_cdot
       complex          ga_cdot_patch
       integer          ga_cluster_nnodes
@@ -42,8 +41,6 @@
       logical          ga_overlay
       integer          ga_pgroup_absolute_id
       integer          ga_pgroup_create
-      integer          ga_pgroup_duplicate
-      integer          ga_pgroup_self
       logical          ga_pgroup_destroy
       integer          ga_pgroup_get_default
       integer          ga_pgroup_get_mirror
@@ -74,7 +71,6 @@
       double complex   ga_zdot
       double complex   ga_zdot_patch
       logical          nga_allocate
-      logical          nga_deallocate
       complex          nga_cdot
       complex          nga_cdot_patch
       integer          nga_cluster_nnodes
@@ -122,8 +118,6 @@
       logical          nga_overlay
       integer          nga_pgroup_absolute_id
       integer          nga_pgroup_create
-      integer          nga_pgroup_duplicate
-      integer          nga_pgroup_self
       logical          nga_pgroup_destroy
       integer          nga_pgroup_get_default
       integer          nga_pgroup_get_mirror
@@ -157,7 +151,6 @@
       double complex   nga_zdot_patch
 !
       external ga_allocate
-      external ga_deallocate
       external ga_cdot
       external ga_cdot_patch
       external ga_cluster_nnodes
@@ -196,8 +189,6 @@
       external ga_overlay
       external ga_pgroup_absolute_id
       external ga_pgroup_create
-      external ga_pgroup_duplicate
-      external ga_pgroup_self
       external ga_pgroup_destroy
       external ga_pgroup_get_default
       external ga_pgroup_get_mirror
@@ -228,7 +219,6 @@
       external ga_zdot
       external ga_zdot_patch
       external nga_allocate
-      external nga_deallocate
       external nga_cdot
       external nga_cdot_patch
       external nga_cluster_nnodes
@@ -279,8 +269,6 @@
       external nga_overlay
       external nga_pgroup_absolute_id
       external nga_pgroup_create
-      external nga_pgroup_duplicate
-      external nga_pgroup_self
       external nga_pgroup_destroy
       external nga_pgroup_get_default
       external nga_pgroup_get_mirror
